cmd/whoami.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/backend/cloud"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+)
+
+func newWhoAmICmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "whoami",
+		Short: "Display current logged in user",
+		Long: "Display current logged in user\n" +
+			"\n" +
+			"Displays the username of the currently logged in user.",
+		Args: cmdutil.NoArgs,
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			b, err := currentBackend()
+			if err != nil {
+				return err
+			}
+
+			cb, ok := b.(cloud.Backend)
```

Comment:
Alternatively, make this part of the backend interface, and have the local one error.  Down the road, I assume we will support other user authentication mechanisms (e.g., AD on-prem), which would be decided in the backend.

Reply:
SGTM